sda-commons-shared-otel-instrumentation/README.md

```diff
@@ -0,0 +1,82 @@
+The configuration can be done either with environment variables or system properties. Please note that system properties will have a higher priority over environment variables.
+
+A full list of the configurable properties can be found in the [autoconfigure module](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure#exporters).
+
+It is recommended to use same configuration across all services in the same domain, to ensure the right context propagation, and export traces to the same monitoring backend.
+
+## Service Name
+
+The service name is used to identify the source (the sender) of the received telemetry data in the monitoring backend. Therefore it's important that every service has a unique name.
+To provide the application with a custom name for each deployment, the `OTEL_SERVICE_NAME` environment variable must be used.
+
+## Basic configuration
+
+By default, the agent exports traces in the otlp format, to an [openTelemetry collector](https://opentelemetry.io/docs/collector/). The most basic configuration would be to provide the service name and a collector endpoint.
+- `OTEL_SERVICE_NAME=my-service`
+- `OTEL_EXPORTER_OTLP_ENDPOINT=http://otel-gateway-host:4317`.
+
+Or in case if the collector is deployed as a sidecar:
+- `OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4317`
+
+## Using Jaeger exporter
+
+To exports traces in Jaeger format, to a [Jaeger collector](https://hub.docker.com/r/jaegertracing/jaeger-collector/). This can be very handy for less local setup overhead, where the [All in one Jaeger image](https://www.jaegertracing.io/docs/1.6/getting-started/#all-in-one-docker-image) can be enough.
+- `OTEL_SERVICE_NAME=my-service`
+- `OTEL_EXPORTER_JAEGER_ENDPOINT=http://jaeger-collector-host:14250`
+
+> **_NOTE:_** It is recommended to not use this option in production even if the used monitoring backend is Jaeger. It is better to export the telemetry data in the _OTLP_ format to a collector, and configure it to handle the transformation.
+
+# Disabling instrumentation
+
+In order to disable tracing in the applications that are using this bundle, or the [starter bundle](../sda-commons-server-starter), the environment variable `OTEL_JAVAAGENT_ENABLED=false` can be used.
+Please note that setting `OTEL_JAVAAGENT_ENABLED` to false will force the bundle to completely skip loading the agent Jar into the JVM, so manual instrumentation like added in the [sda-commons-shared-otel-tracing-example](../sda-commons-shared-otel-tracing-example) will not be possible anymore.
+In this case manual tracing must be done differently (see other options for [manual instrumentation](https://opentelemetry.io/docs/instrumentation/java/manual/)).
+
+# Enable manual instrumentation only
+
+In order to suppress all auto instrumentation but still want to load the agent the option `OTEL_INSTRUMENTATION_COMMON_DEFAULT_ENABLED=false`.
```

Comment:
What could be a valid  scenario for manual instrumentation? When should we recommend it? If this is not desired, we should not mention it IMHO.

Reply:
I also say that this is a very rare case. There would be many reasons for adding manual instrumentation (adding more business attributes, tracing some internal behaviour...), but right now I can not think of any for disabling automatic instrumentation. This was merely a continuation for the previous notes about disabling the agent and how to overcome such limitation
>In this case manual tracing must be done differently (see other options for [manual instrumentation](https://opentelemetry.io/docs/instrumentation/java/manual/)).

If such a case arises, this would describe an easier way to achieve this with the modules that we already have, without having to look for the other options.